src/Elm/Parser/Comments.elm

```diff
@@ -31,8 +32,30 @@ multilineCommentInner : Parser State String
 multilineCommentInner =
     Core.getChompedString (Core.multiComment "{-" "-}" Nestable)
         |> Combine.fromCore
+        |> Combine.backtrackable
+        |> Combine.andThen
+            (\comment ->
+                if String.startsWith "{-|" comment then
+                    Combine.fail "unexpected documentation comment"
```

Comment:
I used another `backtrackable` here, I'm not sure how much it affects performance.